libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -4137,6 +4137,209 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
+                        {
+                            (parsed, error) = ParseTimexProperty(args[0]);
+                        }
+
+                        if (error == null)
+                        {
+                            if (parsed.Year != null || parsed.Month == null || parsed.DayOfMonth == null)
+                            {
+                                error = $"The paramter {args[0]} must be a timex expression only month and day-of-month, for example: XXXX-10-31.";
+                            }
+                        }
+
+                        if (error == null)
+                        {
+                            var (year, month, day) = (DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day);
+                            if (parsed.Month > month || (parsed.Month == month && parsed.DayOfMonth >= day))
+                            {
+                                validYear = year;
+                            }
+                            else
+                            {
+                                validYear = year + 1;
+                            }
+
+                            validMonth = parsed.Month ?? 0;
+                            validDay = parsed.DayOfMonth ?? 0;
+
+                            if (validMonth == 2 && validDay == 29)
+                            {
+                                while (!DateTime.IsLeapYear(validYear))
+                                {
+                                    validYear += 1;
+                                }
+                            }
+
+                            result = TimexProperty.FromDate(new DateTime(validYear, validMonth, validDay)).TimexValue;
+                        }
+
+                        return (result, error);
+                    },
+                    ReturnType.String,
+                    ValidateUnaryString),
+                new ExpressionEvaluator(
+                    ExpressionType.GetPreviousViableDate,
+                    (expr, state, options) =>
+                    {
+                        TimexProperty parsed = null;
+                        string result = null;
+                        string error = null;
+                        var validYear = 0;
+                        var validMonth = 0;
+                        var validDay = 0;
+                        IReadOnlyList<object> args;
+                        (args, error) = EvaluateChildren(expr, state, options);
+                        if (error == null)
+                        {
+                            (parsed, error) = ParseTimexProperty(args[0]);
+                        }
+
+                        if (error == null)
+                        {
+                            if (parsed.Year != null || parsed.Month == null || parsed.DayOfMonth == null)
+                            {
+                                error = $"The paramter {args[0]} must be a timex expression only contains month and day-of-month, for example: XXXX-10-31.";
```

Comment:
typo on "parameter". 

The grammer on the error message is problematic. Please revise.

Reply:
changed